### PR TITLE
feat(dpmodel): support Array API learning rate

### PR DESCRIPTION
**Pull request by njzjz**
This is useful when the LR is within the JAX JIT compilation - the step is given as a `jnp.ndarray` and NumPy should not be used.

## Summary by CodeRabbit

* **New Features**
  * Learning-rate API now accepts array-like steps and computes schedules using backend-agnostic array operations; exponential and cosine schedulers support array inputs and preserve minimum learning-rate behavior.

* **Tests**
  * Added cross-backend consistency tests to validate identical learning-rate outputs across NumPy, PyTorch, JAX and array-api backends.

* **Chores**
  * Updated test dependency constraint and bumped default test API version used by strict array-api tests.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: source/tests/consistent/test_learning_rate.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        np.testing.assert_allclose(self.ref, to_numpy_array(test), atol=1e-10)
+
+    @unittest.skipUnless(INSTALLED_PT, "PyTorch is not installed")
+    def test_pt_consistent_with_ref(self) -> None:
+        with torch.device("cpu"):
+            self.compare_test_with_ref(torch_xp)
+
+    @unittest.skipUnless(
+        INSTALLED_ARRAY_API_STRICT, "array_api_strict is not installed"
+    )
```
[the remark this answers is not in the document]
Comment (njzjz) [reply source/tests/consistent/test_learning_rate.py:75]: Fixed.

File: deepmd/dpmodel/utils/learning_rate.py [hunk 2]
```diff
@@ -88,16 +88,19 @@ def __init__(
...
         self.decay_rate = np.exp(
             np.log(stop_lr / self.start_lr) / (stop_steps / self.decay_steps)
-        )
+        ).item()
         if decay_rate is not None:
             self.decay_rate = decay_rate
         self.min_lr = self.stop_lr
 
-    def value(self, step: int) -> np.float64:
+    def value(self, step: int, xp: Any = np) -> np.float64:
```
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/dpmodel/utils/learning_rate.py]: Fixed.

**Review by iProzd — APPROVED** [review]
